### PR TITLE
args check in book list APIs

### PR DESCRIPTION
**Pull request by AcepcsMa**
Perform args check in book list APIs.

### CODE REVIEW

File: application/API/booklist.py [hunk 1]
```diff
@@ -23,8 +23,19 @@ class BookLists(Resource):
...
+            return {
+                'message': 'Please provide your user id!'
+            }, 400
+
+        try:
+            list_name = args['list_name']
+        except:
+            return {
+                'message': 'Please provide the book list name!'
+            }, 400
```
Comment (NEUanransu) [inline application/API/booklist.py:39]: Maybe also do something like 
if user_id is None: return 400 code ?

File: application/API/booklist.py [hunk 2]
```diff
@@ -23,8 +23,19 @@ class BookLists(Resource):
...
 
         books = db.session.query(Book) \
             .join(BookListToBook, Book.id == BookListToBook.book_id) \
```
Comment (NEUanransu) [inline application/API/booklist.py:52]: Maybe also check if there exist a BookList with the user_id and booklist_name?
Do something like:
if not books: return error code and "no mathcing booklist"

File: application/API/booklist.py [hunk 3]
```diff
@@ -66,32 +94,76 @@ def delete(self):
...
-        user_id = args['user_id']
-        list_name = args['list_name']
-        description = args['description']
+        try:
+            user_id = args['user_id']
+        except:
+            return {
+                'message': 'Please provide your user id!'
+            }, 401
+
```
Comment (NEUanransu) [inline application/API/booklist.py:114]: Also check if the User with user_id exist in the DB?
Do something like:

user = db.query(User).filter(User.id == user_id).first()
if  user is None: return 400 code and "no such user"

File: application/API/booklist.py [hunk 4]
```diff
@@ -118,9 +190,26 @@ def delete(self):
...
+            return {
+                'message': 'Please provide the book list name!'
+            }, 400
+
+        try:
+            book_id = args['book_id']
+        except:
+            return {
+                'message': 'Please provide the book id!'
+            }, 400
```
Comment (NEUanransu) [inline application/API/booklist.py:232]: Need to check if the User of user_id and Book of book_id exist

**Review by NEUanransu — CHANGES_REQUESTED** [review]
See single comments. Need to handle more error cases

File: application/API/booklist.py (hunk 1 above)
Comment (AcepcsMa) [reply application/API/booklist.py:39]: > Maybe also do something like
> if user_id is None: return 400 code ?

No, there is no need to do that. When there is an exception being caught, it means that the user_id does NOT exist.

**Review by NEUanransu — APPROVED** [review]

**Pull request merged**